include/fastdds/topic/qos/TopicQos.hpp

```diff
@@ -0,0 +1,118 @@
+// Copyright 2019 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file TopicQos.hpp
+ */
+
+
+#ifndef _FASTDDS_TOPICQOS_HPP
+#define _FASTDDS_TOPICQOS_HPP
+
+#include "../../../fastrtps/qos/QosPolicies.h"
+
+namespace eprosima {
+namespace fastdds {
+
+/**
+ * Class TopicQos, containing all the possible Qos that can be set for a determined Topic.
+ * Although these values can be set and are transmitted
+ * during the Endpoint Discovery Protocol, not all of the behaviour associated with them has been implemented in the library.
+ * Please consult each of them to check for implementation details and default values.
+ * @ingroup FASTRTPS_ATTRIBUTES_MODULE
+ */
+class TopicQos
+{
+public:
+    RTPS_DllAPI TopicQos();
+    RTPS_DllAPI virtual ~TopicQos();
+
+    bool operator==(const TopicQos& b) const
+    {
+        return (this->m_topicData == b.m_topicData) &&
+               (this->m_durability == b.m_durability) &&
+               (this->m_durabilityService == b.m_durabilityService) &&        
+               (this->m_deadline == b.m_deadline) &&        
+               (this->m_latencyBudget == b.m_latencyBudget) &&        
+               (this->m_liveliness == b.m_liveliness) &&        
+               (this->m_reliability == b.m_reliability) &&        
+               (this->m_destinationOrder == b.m_destinationOrder) &&        
+               (this->m_history == b.m_history) &&        
+               (this->m_resourceLimits == b.m_resourceLimits) &&        
+               (this->m_transportPriority == b.m_transportPriority) &&        
+               (this->m_lifespan == b.m_lifespan) &&        
+               (this->m_ownership == b.m_ownership);
+    }
+
+    //!Topic Data Qos, NOT implemented in the library.
+    fastrtps::TopicDataQosPolicy m_topicData;
```

Comment:
I know you probably copied this from an existing Qos struct, but from now on, please follow the coding style: https://github.com/richiware/cppstyle/blob/master/README.es.md both for member variables and methods